src/btree/row_srch.c

```diff
@@ -197,6 +204,66 @@ __wt_search_insert(
     cbt->compare = -cmp;
     cbt->ins = (ins != NULL) ? ins : last_ins;
     cbt->ins_head = ins_head;
+
+    /*
+     * FIXME WT-10439 - Available categories are changing in WT-10439. Whichever change merges
+     * second needs to change this to WT_DIAGNOSTIC_BTREE_VALIDATE.
+     */
+    if (EXTRA_DIAGNOSTICS_ENABLED(session, WT_DIAG_DATA_VALIDATION))
+        WT_RET(__validate_next_stack(session, cbt->next_stack));
+    return (0);
+}
+
+/*
+ * __validate_next_stack --
+ *     Verify that for each level in the provided next_stack that higher levels on the stack point
+ *     to larger inserts than lower levels.
+ */
+static inline int
+__validate_next_stack(WT_SESSION_IMPL *session, WT_INSERT *next_stack[WT_SKIP_MAXDEPTH])
```

Comment:
Can we add a final validation that next_stack[0] is larger than the search key.

Reply:
Added in [17bae81](https://github.com/wiredtiger/wiredtiger/pull/8782/commits/17bae818285f1eae39e2c0ad84dff6ede5b0b8d1)